### PR TITLE
Feature/115 data init

### LINKED ISSUE #115: Move data-init to resource loader
Currently, initialising plugins is the responsibility of biotope-boilerplate.

Since resource-loader is the technology providing support for that kind functionality in the first place, it should part of this codebase.

### PR DESCRIPTION
**Pull request by ctmm**
Closes #115 

- Fix: Resources ready event;
- Feature: Add register script helper;
- Feature: Add initialise script after loaded;
- Chore: Increase test coverage;
- Chore: Fix dependency security issues;

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -30,3 +30,66 @@ const resourceLoader = new ResourceLoader({
 ```
 
 To know what data should go inside the resources attribute, have a look at the [API documentation](./api.md)
+
+## Initializing plugins
+By default, the resource-loader will try to initialize loaded resources. 
+To set which plugin to initialize you'll need to register the component with a name and pass that same name via attributes in your html code.
+Please check below for more info on the necessary steps:
+
+1. Register the component, using the `registerPlugin` helper from the Resource Loader module:
```
[the remark this answers is not in the document]
Comment (SheepFromHeaven) [reply docs/usage.md]: @timomayer As this is not only reserved to components I would prefer just a plain `registerInitScript`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@biotope/resource-loader",
-  "version": "2.1.1",
+  "version": "3.0.0",
```
[the remark this answers is not in the document]
Comment (SheepFromHeaven) [reply package.json]: please do not manually bump the versions, that's what the release script will do :/

[the remark this answers is not in the document]
Comment (ctmm) [reply package.json]: Sorry, I was unaware of that. Will fix!

File: docs/usage.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctmm) [reply docs/usage.md]: I understand the point.

However, `registerInitScript` comes across as not exactly on point to me.
The helper is not initialising. It's adding an "address" through which that logic can be accessed later on.

How about `registerScript`?

[the remark this answers is not in the document]
Comment (ctmm) [reply docs/usage.md]: I assume renaming should extend to the related resource loader options so as to make their relationship more obvious to users.

```javascript
export interface ResourceLoaderOptions {
    …
    initScripts?: boolean;
    initScriptAttributeSelector?: string;
    scriptOptionsAttributeSelector?: string;
    …
}
```

**Comment by ctmm** [comment]
@timomayer @SheepFromHeaven 
Thank you for your insights. I have updated the code to reflect the issues you raised.
Kindly ask you for another round.

Thanks

**Comment by timomayer** [comment]
@ctmm please go ahead and merge :)

**Pull request merged**